casper/src/test/scala/coop/rchain/casper/batch2/ValidateTest.scala

```diff
@@ -599,48 +498,45 @@ class ValidateTest
             ) shouldBeF false
       } yield ()
   }
 
-  "Block hash format validation" should "fail on invalid hash" in withStorage {
-    _ => implicit blockDagStorage =>
-      val context  = buildGenesis()
-      val (sk, pk) = context.validatorKeyPairs.head
-      val sender   = ByteString.copyFrom(pk.bytes)
-      for {
-        _                <- blockDagStorage.insert(genesis, false, approved = true)
-        dag              <- blockDagStorage.getRepresentation
-        latestMessageOpt <- dag.latestMessage(sender)
-        seqNum           = latestMessageOpt.fold(0)(_.seqNum) + 1
-        genesis = ValidatorIdentity(sk)
-          .signBlock(context.genesisBlock.copy(seqNum = seqNum))
-        _ <- Validate.blockHash[Task](genesis) shouldBeF true
-        result <- Validate.blockHash[Task](
-                   genesis.copy(blockHash = ByteString.copyFromUtf8("123"))
-                 ) shouldBeF false
-      } yield result
+  "Block hash format validation" should "fail on invalid hash" in {
+    implicit val aBlock = arbBlockMessage
+
+    forAll { (block: BlockMessage) =>
+      val hash           = ProtoUtil.hashBlock(block)
+      val blockValidHash = block.copy(blockHash = hash)
+
+      // Test valid block hash
+      val hashValid = Validate.blockHash[Task](blockValidHash).runSyncUnsafe()
+
+      hashValid shouldBe true
+
+      val blockInValidHash = block.copy(blockHash = ByteString.copyFromUtf8("123"))
+
+      // Test invalid block hash
+      val hashInValid = Validate.blockHash[Task](blockInValidHash).runSyncUnsafe()
+
+      hashInValid shouldBe false
+    }
   }
 
-  "Block version validation" should "work" in withStorage { _ => implicit blockDagStorage =>
-    val context  = buildGenesis()
-    val (sk, pk) = context.validatorKeyPairs.head
-    val sender   = ByteString.copyFrom(pk.bytes)
-    for {
-      _                <- blockDagStorage.insert(genesis, false, approved = true)
-      dag              <- blockDagStorage.getRepresentation
-      latestMessageOpt <- dag.latestMessage(sender)
-      seqNum           = latestMessageOpt.fold(0)(_.seqNum) + 1
-      genesis = ValidatorIdentity(sk).signBlock(
-        context.genesisBlock.copy(seqNum = seqNum)
-      )
-      _      <- Validate.version[Task](genesis, -1) shouldBeF false
-      result <- Validate.version[Task](genesis, 1) shouldBeF true
-    } yield result
+  "Block version validation" should "allow supported versions" in {
+    implicit val aBlock = arbBlockMessage
+
+    forAll { (block: BlockMessage, version: Int) =>
+      val blockWithVersion = block.copy(version = version)
+
+      // Expected one of hard-coded block versions supported by this version of RNode software
+      val expectedValid = BlockVersion.Supported.contains(version)
+      // Actual validation
+      val actualValid = Validate.version[Task](blockWithVersion).runSyncUnsafe()
+
+      actualValid shouldBe expectedValid
+    }
```

Comment:
Most of the validation tests should be rewritten as generative tests like these without creation of additional data not necessary for testing.